pkg/latte/intent_change_email.go

```diff
@@ -0,0 +1,87 @@
+package latte
+
+import (
+	"context"
+
+	"github.com/authgear/authgear-server/pkg/api"
+	"github.com/authgear/authgear-server/pkg/api/apierrors"
+	"github.com/authgear/authgear-server/pkg/api/model"
+	"github.com/authgear/authgear-server/pkg/lib/authn/identity"
+	"github.com/authgear/authgear-server/pkg/lib/session"
+	"github.com/authgear/authgear-server/pkg/lib/workflow"
+	"github.com/authgear/authgear-server/pkg/util/validation"
+)
+
+func init() {
+	workflow.RegisterPublicIntent(&IntentChangeEmail{})
+}
+
+var IntentChangeEmailSchema = validation.NewSimpleSchema(`
+	{
+		"type": "object",
+		"additionalProperties": false
+	}
+`)
+
+type IntentChangeEmail struct {
```

Comment:
This intent merely updates the identity, but it misses a lot of details. Please take a look at pkg/lib/interaction/intents/update_identity.go and pkg/latte/intent_create_login_id.go.

We have a few things to do

1. Call UpdateWithSpec
2. Call CheckDuplicate
3. Call Update
4. Populate standard attributes
5. Verify the identity
6. Create the OOB OTP authenticator
7. Fire the event produced by NewIdentityLoginIDUpdatedEventPayload

intent_create_login_id.go has done 4, 5, 6
update_identity.go has done 1, 2, 3, 7

Your workflow should do all of them

Reply:
Great! Thank you.